pypeit/core/coadd.py

```diff
@@ -766,19 +799,36 @@ def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
             weights[:, iexp] = sn_conv
     else:
         weights = np.zeros_like(flux_stack)
+        refscale = (sn_val[:, ref_spec] > 0) / (sn_val[:, ref_spec] + (sn_val[:, ref_spec] == 0))
         for iexp in range(nstack):
+            if relative_weights:
```

Comment:
Please change the control flow here to be 

if ivar_weights: 

elif relative_weights:

else: 
old code which was the default. 

Otherwise the control flow is impossible to follow with all of these if relative weights statements inserted. If this means that you have to do the loop over exposures twice so be it.